### PR TITLE
Add unit tests & related functionality for emit i32 delta

### PR DESCRIPTION
**Pull request by ameykasbe**

### CODE REVIEW

File: tsz-compress/src/v2/halfvec.rs [hunk 1]
```diff
@@ -185,6 +185,25 @@ impl HalfVec {
     }
 }
 
+/// Compares two HalfVec for equality.
+impl PartialEq for HalfVec {
```
Comment (trueb2) [inline tsz-compress/src/v2/halfvec.rs]: This is not a valid implementation of PartialEq. We would not want to provide this as a library feature due to the expense of a correct check.

If this is for use in tests, then we can provide a function in the tests module.

File: tsz-compress/src/v2/halfvec.rs [hunk 2]
```diff
@@ -185,6 +185,25 @@ impl HalfVec {
...
+    fn eq(&self, other: &Self) -> bool {
+        self.words == other.words && self.len == other.len
+    }
+}
+
+/// Compares two HalfWord for equality.
+impl PartialEq for HalfWord {
```
Comment (trueb2) [inline tsz-compress/src/v2/halfvec.rs]: Similar to above, we do not have a use case for this within the library as far as I know.

**Review by trueb2 — CHANGES_REQUESTED** [review]

File: tsz-compress/src/v2/halfvec.rs (hunk 1 above)
Comment (ameykasbe) [reply tsz-compress/src/v2/halfvec.rs]: Changes in commit [c19be9f](https://github.com/qsib-cbie/tsz/pull/36/commits/c19be9f42ebca7fe539b84512e03f6fb671f51db).

File: tsz-compress/src/v2/halfvec.rs (hunk 2 above)
Comment (ameykasbe) [reply tsz-compress/src/v2/halfvec.rs]: Changes in commit [c19be9f](https://github.com/qsib-cbie/tsz/pull/36/commits/c19be9f42ebca7fe539b84512e03f6fb671f51db).

**Review by trueb2 — APPROVED** [review]

**Pull request merged**